src/librustdoc/passes/collect_trait_impls.rs

```diff
@@ -126,36 +127,32 @@ crate fn collect_trait_impls(mut krate: Crate, cx: &mut DocContext<'_>) -> Crate
         }
     });
 
-    // `tcx.crates(())` doesn't include the local crate, and `tcx.all_trait_implementations`
-    // doesn't work with it anyway, so pull them from the HIR map instead
-    let mut extra_attrs = Vec::new();
-    for trait_did in cx.tcx.all_traits() {
-        for &impl_did in cx.tcx.hir().trait_impls(trait_did) {
-            let impl_did = impl_did.to_def_id();
-            cx.tcx.sess.prof.generic_activity("build_local_trait_impl").run(|| {
```

Comment:
Would it be possible to keep this prof block? It looks like it may be useful, given that I've never actually played with the profiling.

Reply:
Added a couple more timers to cover all calls to `build_impl`.